### PR TITLE
Assign components for bugs in the 'General' component too

### PR DESCRIPTION
**Pull request by marco-c**
For now only recent bugs and with a higher confidence threshold

### CODE REVIEW

File: auto_nag/scripts/component.py [hunk 1]
```diff
@@ -73,22 +84,31 @@ def get_bugs(self, date='today', bug_ids=[]):
...
+        for bug, prob, index, suggestion in zip(bugs, probs, indexes, suggestions):
             # Skip product-only suggestions that are not useful.
-            if '::' not in component and bug['product'] == component:
+            if '::' not in suggestion and bug['product'] == suggestion:
                 continue
 
-            component = self.model.CONFLATED_COMPONENTS_MAPPING.get(component, component)
+            suggestion = self.model.CONFLATED_COMPONENTS_MAPPING.get(suggestion, suggestion)
+
+            assert '::' in suggestion
```
Comment (calixteman) [inline auto_nag/scripts/component.py]: Suggestion: you could use logger.error(...) and have an error report and then avoid to break everything if one suggestion doesn't contain '::'

File: auto_nag/scripts/component.py [hunk 2]
```diff
@@ -73,22 +84,31 @@ def get_bugs(self, date='today', bug_ids=[]):
...
+            i = suggestion.index('::')
+            suggested_product = suggestion[:i]
+            suggested_component = suggestion[i + 2:]
+
+            # When moving bugs out of the 'General' component, we don't want to change the product (unless it is Firefox).
+            if bug['component'] == 'General' and bug['product'] != suggested_product and bug['product'] != 'Firefox':
```
Comment (calixteman) [inline auto_nag/scripts/component.py]: nit: bug['product'] not in {suggested_product, 'Firefox'}

File: auto_nag/scripts/component.py [hunk 3]
```diff
@@ -43,15 +43,26 @@ def sort_columns(self):
...
+
+            # Ignore closed bugs.
             'bug_status': '__open__',
-        }
 
-        return params
+            # Get recent General bugs, and all Untriaged bugs.
+            'j3': 'OR',
+            'f3': 'OP',
+                'f4': 'OP',
```
Comment (calixteman) [inline auto_nag/scripts/component.py]: did you check that AND is the default operator for f4 ?

**Comment by sylvestre** [comment]
auto_nag/scripts/component.py:59:17: E131 continuation line unaligned for hanging indent
auto_nag/scripts/component.py:60:21: E131 continuation line unaligned for hanging indent
=> could you please fix that?

File: auto_nag/scripts/component.py (hunk 2 above)
Comment (marco-c) [reply auto_nag/scripts/component.py]: I was thinking about that, but I was not sure which one would be more readable in this particular case

File: auto_nag/scripts/component.py (hunk 1 above)
Comment (marco-c) [reply auto_nag/scripts/component.py]: If '::' is not in the suggestion, something must be broken in bugbug.
If this breaks all the rest of the checkers though, I'll just report an error.

**Comment by marco-c** [comment]
> could you please fix that?

I'll add some # noqa, I used a wrong indentation on purpose here to "show" the parentheses of the query

File: auto_nag/scripts/component.py (hunk 3 above)
Comment (marco-c) [reply auto_nag/scripts/component.py]: Yep, I think AND is always the default.
I'll put 'j4' just to make this more readable.

Comment (calixteman) [reply auto_nag/scripts/component.py]: I think you're right (about AND is the default) but just in case...

**Pull request merged**